weaviate/collection/data.py

```diff
@@ -326,19 +324,30 @@ def __parse_properties_grpc(data: Dict[str, Any]) -> weaviate_pb2.BatchObject.Pr
         )
 
 
-class _DataCollection(Generic[Properties], _Data):
+class _DataCollection(Generic[TProperties], _Data):
     def __init__(
         self,
         connection: Connection,
         name: str,
-        config: _ConfigBase,
         consistency_level: Optional[ConsistencyLevel],
         tenant: Optional[str],
-        type_: Optional[Type[Properties]] = None,
+        type_: Optional[Type[TProperties]] = None,
     ):
-        super().__init__(connection, name, config, consistency_level, tenant)
+        super().__init__(connection, name, consistency_level, tenant)
         self.__type = type_
 
+    def with_data_model(self, data_model: Type[Properties]) -> "_DataCollection[Properties]":
+        if data_model is not None and get_origin(data_model) is not dict:
+            try:
```

Comment:
why not isinstance(...)?

Reply:
Because it expects a type argument, e.g. MyProps, and not an object, e.g. MyProps(). I think `isinstance` may work when using classes that inherit from `dataclass` or `pydantic.BaseModel` but because `typing.TypedDict` is a special kind of class it doesn't work with `isinstance` at all

This is a bit of a hacky-workaround that is saying: "If `data_model` is either a raw `dict` or a `typing.TypedDict`"

I'll add a comment here making this clear